sql-plugin/src/main/spark350/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -340,4 +384,145 @@ object GpuDataWritingSparkTask extends GpuWritingSparkTask[DataWriter[ColumnarBa
   override protected def write(writer: DataWriter[ColumnarBatch], row: ColumnarBatch): Unit = {
     writer.write(row)
   }
+}
+
+/**
+ * GPU version of DeltaWritingSparkTask for writing delta (position delete) operations.
+ * Applies projections to extract row data and metadata before writing.
+ */
+case class GpuDeltaWritingSparkTask(
+    projs: WriteDeltaProjections) extends GpuWritingSparkTask[DeltaWriter[ColumnarBatch]] {
+
+  private lazy val rowProjection = projs.rowProjection.map(GpuProjectingColumnarBatch(_)).orNull
+  private lazy val rowDataTypes = projs.rowProjection
+    .map(_.schema.fields.map(f => f.dataType))
+    .orNull
+
+  private lazy val rowIdProjection = GpuProjectingColumnarBatch(projs.rowIdProjection)
+  private lazy val rowIdDataTypes = rowIdProjection.schema.fields.map(_.dataType)
+
+  override protected def write(writer: DeltaWriter[ColumnarBatch], batch: ColumnarBatch): Unit = {
+    withRetryNoSplit(batch) { _ =>
+      val deleteFilter = filterByOperation(batch, DELETE_OPERATION)
+      withResource(deleteFilter) { _ =>
+        withResource(rowIdProjection.project(batch)) { rowIds =>
+          val rowIdBatch = GpuColumnVector.filter(rowIds, rowIdDataTypes, deleteFilter)
+          writer.delete(null, rowIdBatch)
+        }
+      }
+
+      val updateFilter = filterByOperation(batch, UPDATE_OPERATION)
+      withResource(updateFilter) { _ =>
+        val rowIds =  withResource(rowIdProjection.project(batch)) { rowIds =>
+          GpuColumnVector.filter(rowIds, rowIdDataTypes, updateFilter)
+        }
+
+        closeOnExcept(rowIds) { _ =>
+          val rows = withResource(rowProjection.project(batch)) { rows =>
+            GpuColumnVector.filter(rows, rowDataTypes, updateFilter)
+          }
+
+          writer.update(null, rowIds, rows)
```

Comment:
skip if `numRows == 0` ?
I saw `GpuDeltaWithMetadataWritingSparkTask` has this kind of logic.